### PR TITLE
Select session for visualization

### PR DESCRIPTION
**Pull request by XX-Yin**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "production_testing" into "main" should use the keyword "production merge" in the title for reliable indexing of updates
  
### Describe changes:
Added a combo box and spin box to show all sessions related to that animal. We can select a specific session for visualization. It's useful for navigating different sessions of one animal. 
image
image

### Describe the expected change in behavior from the perspective of the experimenter
No

### Describe the outcome of testing this update on a rig in 447
- [x] test on my own computer
- [x] test in ephys

### CODE REVIEW

**Comment by hanhou** [comment]
This is a useful addition, although it somewhat overlaps with my latest PR #207. The drawbacks of doing such visualization in the GUI include:
1. you can only load local sessions
2. the interactivity is very limited
3. duplicated development efforts
4. it goes beyond what a GUI is supposed to do.

I think linking to the app (or any visualization tools in the future) is much more flexible. If you'd like, I can show you how to replicate exactly the GUI's plot (with all lick time etc.) in Plotly and put it to the Streamlit app.

**Comment by XX-Yin** [comment]
> Why did you only update one of the two `.ui` files?

It's not finished. I will finish it today. Would it be better to delete the "[Do Not Merge]" (indicating finish of the code) in the title before reviewing the PR in the future?

**Comment by hanhou** [comment]
> > Why did you only update one of the two `.ui` files?
> 
> It's not finished. I will finish it today. Would it be better to delete the "[Do Not Merge]" (indicating finish of the code) in the title before reviewing the PR in the future?

How about use "[Do Not Review]", or just make it as a Draft?

**Comment by XX-Yin** [comment]
> > > Why did you only update one of the two `.ui` files?
> > 
> > 
> > It's not finished. I will finish it today. Would it be better to delete the "[Do Not Merge]" (indicating finish of the code) in the title before reviewing the PR in the future?
> 
> How about use "[Do Not Review]", or just make it as a Draft?

That sounds good!

**Comment by XX-Yin** [comment]
> This is a useful addition, although it somewhat overlaps with my latest PR #207. The drawbacks of doing such visualization in the GUI include:
> 
> 1. you can only load local sessions
> 2. the interactivity is very limited
> 3. duplicated development efforts
> 4. it goes beyond what a GUI is supposed to do.
> 
> I think linking to the app (or any visualization tools in the future) is much more flexible. If you'd like, I can show you how to replicate exactly the GUI's plot (with all lick time etc.) in Plotly and put it to the Streamlit app, which byitself is a good addition to our behavior analysis pipeline.

I agree it's better to use centralized visualization.

**Comment by XX-Yin** [comment]
@hanhou and @alexpiet This is ready for review again.

**Comment by hanhou** [comment]
@XX-Yin I changed the base to `develop`, but now there are some conflicts. Can you resolve them locally?

**Comment by XX-Yin** [comment]
> @XX-Yin I changed the base to `develop`, but now there are some conflicts. Can you resolve them locally?

@alexpiet 
I resolved the conflict, but it seems that there is a bug in the `develop`. at line 2557 
image

**Comment by alexpiet** [comment]
> > @XX-Yin I changed the base to `develop`, but now there are some conflicts. Can you resolve them locally?
> 
> @alexpiet I resolved the conflict, but it seems that there is a bug in the `develop`. at line 2557 image

fixed by https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/287

File: src/foraging_gui/Foraging.py [hunk 1]
```diff
@@ -2049,50 +2121,76 @@ def _Open_getListOfMice(self):
...
-        # Start new session
-        new_session = self._NewSession()
-        if not new_session:
-            return
-
-        if open_last:
-            mice = self._Open_getListOfMice()
-            W = MouseSelectorDialog(self, mice)
+    def _Open(self,open_last = False,input_file = ''):
+        if input_file == '' or input_file == False:
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:2125]: What is input_file? What is it sometimes a string, and sometimes a boolean?

Comment (XX-Yin) [reply src/foraging_gui/Foraging.py]: It's a history issue. I deleted open_last = False

**Review by alexpiet — APPROVED** [review]

**Pull request merged**